### PR TITLE
Feature/meeting request v2

### PR DESCRIPTION
**Pull request by MaxCastelCast**
Add the groundwork for the second verision of the Meeting request system :
- The profiles store the uid of hte user to whom they sent a request
- The profiles store a map (uid -> message) that stores the messages received by the user
- backend for messaging system implemented : when a user sends a message, they store uid receiver and the receiver gets (uidSender, message)
- simplified part of the messaging system by removing useless parts
- Addapted the tests to the new version of the messaging system

Problem : 
-I did not had the time to test my code and part of the code (like MeetingRequestService) is untestable due to what we talked about before. 
- This implementation is only backend, nothing ll be showable in the coache meeting
- had to comment out part of the end to end M2 because the messaging workflow is changing
- removed test in MeetingRequestService, could add them back later on and they weren't testing a lot anyway

### CODE REVIEW

**Comment by MaxCastelCast** [comment]
The idea ll be to add the tests later on when i finalize the meeting request messaging system, but i had no time to test this week, sorry, i had a lot of work this week

File: app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestService.kt [hunk 1]
```diff
@@ -24,10 +25,18 @@ class MeetingRequestService : FirebaseMessagingService() {
...
   override fun onMessageReceived(remoteMessage: RemoteMessage) {
     super.onMessageReceived(remoteMessage)
-    remoteMessage.notification?.let { notification ->
-      val title = notification.title ?: "Default Title"
-      val body = notification.body ?: "Default Body"
-      showNotification(title, body)
+    Log.d("MEETING REQUEST", "received stuff")
+    val senderUid = remoteMessage.data["senderUID"] ?: "null"
+    val message = remoteMessage.data["message"] ?: "null"
+    val title = remoteMessage.data["title"]
```
Comment (arthur-mrgt) [inline app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestService.kt:31]: Nitpick: Even if the == operator is null-safe, adding here a null check for consistency would be better

File: app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestService.kt [hunk 2]
```diff
@@ -24,10 +25,18 @@ class MeetingRequestService : FirebaseMessagingService() {
...
+    if (title == "MEETING REQUEST") {
+      Log.d("MEETING REQUEST", "RECEIVED MEETING REQUEST")
+      val meetingRequest = MeetingRequest(message = message, senderUID = senderUid)
+      MeetingRequestManager.meetingRequestViewModel?.addToMeetingRequestInbox(meetingRequest)
+    } else if (title == "MEETING RESPONSE") {
+      Log.d("MEETING RESPONSE", "RECEIVED MEETING RESPONSE")
+    } else if (title == "MEETING CONFIRMATION") {
+      Log.d("MEETING CONFIRMATION", "RECEIVED MEETING CONFIRMATION")
```
Comment (arthur-mrgt) [inline app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestService.kt:39]: Question: You do nothing for theses too cases ?

File: app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestService.kt [hunk 3]
```diff
@@ -24,10 +25,18 @@ class MeetingRequestService : FirebaseMessagingService() {
...
-      val body = notification.body ?: "Default Body"
-      showNotification(title, body)
+    Log.d("MEETING REQUEST", "received stuff")
+    val senderUid = remoteMessage.data["senderUID"] ?: "null"
+    val message = remoteMessage.data["message"] ?: "null"
+    val title = remoteMessage.data["title"]
+    if (title == "MEETING REQUEST") {
+      Log.d("MEETING REQUEST", "RECEIVED MEETING REQUEST")
+      val meetingRequest = MeetingRequest(message = message, senderUID = senderUid)
+      MeetingRequestManager.meetingRequestViewModel?.addToMeetingRequestInbox(meetingRequest)
```
Comment (arthur-mrgt) [inline app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestService.kt:35]: Question: Do you handle possible errors here ?

File: app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestService.kt [hunk 4]
```diff
@@ -24,10 +25,18 @@ class MeetingRequestService : FirebaseMessagingService() {
    */
   override fun onMessageReceived(remoteMessage: RemoteMessage) {
```
Comment (arthur-mrgt) [inline app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestService.kt:26]: Notice:  The if-else blocks for titles could be replaced with a cleaner structure, such as a mapping table using a when statement : 

when (title) {
    MEETING_REQUEST -> {
        Log.d("MEETING REQUEST", "RECEIVED MEETING REQUEST")
        val meetingRequest = MeetingRequest(message = message, senderUID = senderUid)
        MeetingRequestManager.meetingRequestViewModel?.addToMeetingRequestInbox(meetingRequest)
    }
    MEETING_RESPONSE -> Log.d("MEETING RESPONSE", "RECEIVED MEETING RESPONSE")
    MEETING_CONFIRMATION -> Log.d("MEETING CONFIRMATION", "RECEIVED MEETING CONFIRMATION")
    else -> Log.w("MESSAGE RECEIVED", "Unknown title: $title")
}

File: app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestViewModel.kt [hunk 5]
```diff
@@ -82,10 +75,6 @@ class MeetingRequestViewModel(
   fun onMeetingRequestChange(newMessage: String) {
     meetingRequestState = meetingRequestState.copy(message = newMessage)
   }
-  /** Sets the status of the message, shows that it is ready to be sent */
-  fun onSubmitMeetingRequest() {
-    meetingRequestState = meetingRequestState.copy(isEnteringMessage = false)
-  }
 
   /** Sends the message from our user to the target user */
   fun sendMessage() {
```
Comment (arthur-mrgt) [inline app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestViewModel.kt:80]: Important: Both sendMessage and sendMeetingRequest perform nearly identical tasks, differing only by the Firebase Function name. This duplication could be refactored into a single reusable method.

File: app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestViewModel.kt [hunk 6]
```diff
@@ -95,19 +84,74 @@ class MeetingRequestViewModel(
...
+                .await()
+
+        meetingRequestState = meetingRequestState.copy(message = "")
+      } catch (e: Exception) {
+        Log.e("FIREBASE ERROR", "Error sending message", e)
+      }
+    }
+  }
+
+  fun addToMeetingRequestSent(receiverUID: String) {
```
Comment (arthur-mrgt) [inline app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestViewModel.kt:124]: Nitpick: Methods like addToMeetingRequestSent and addToMeetingRequestInbox use profilesViewModel.getProfileByUidAndThen, which relies on the selected profile’s value being set. This indirect dependency could introduce race conditions if the value is updated elsewhere.

File: app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestViewModel.kt [hunk 7]
```diff
@@ -95,19 +84,74 @@ class MeetingRequestViewModel(
...
+              meetingRequestInbox = currentMeetingRequestInbox + (m.senderUID to m.message))
+      if (updatedProfile != null) {
+        profilesViewModel.updateProfile(updatedProfile)
+      } else {
+        Log.e("INBOX MEETING REQUEST", "Adding the new meeting request to our inbox list failed")
+      }
+    }
+  }
+
+  fun getInbox() {}
```
Comment (arthur-mrgt) [inline app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestViewModel.kt:156]: Question: Not implemented yet ?

File: app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt [hunk 8]
```diff
@@ -220,6 +223,23 @@ open class ProfilesViewModel(
         onFailure = { e -> handleError(e) })
   }
 
+  fun getInboxByUidAndThen(uidList: List<String>, andThen: () -> Unit) {
+    _loading.value = true
+    _inbox.value = listOf()
+    for (uid in uidList) {
+      Log.d("GET UID", uid)
```
Comment (arthur-mrgt) [inline app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt:230]: Useless log

**Review by arthur-mrgt — CHANGES_REQUESTED** [review]
Overall, a good PR, but please pay attention to the following points:

- Code Duplication: Try to reuse existing code as much as possible to avoid redundancy.
- Error Handling: Explicitly handle errors when necessary, instead of simply logging them.
- Modularity and Flexibility: Aim for modular and flexible code that can adapt to future updates. The current code is quite rigid with many hardcoded values.

As discussed, tests will need to be added to achieve satisfactory coverage.

**Review by arthur-mrgt — APPROVED** [review]
Looks good to me.

**Pull request merged**